diff --git a/Tables/dbo.RJTEst.sql b/Tables/dbo.RJTEst.sql
deleted file mode 100644
index 2900e5d..0000000
--- a/Tables/dbo.RJTEst.sql
+++ /dev/null
@@ -1,9 +0,0 @@
-CREATE TABLE [dbo].[RJTEst]
-(
-[a] [float] NOT NULL,
-[b] [float] NOT NULL,
-[c] [int] NULL
-) ON [PRIMARY]
-GO
-ALTER TABLE [dbo].[RJTEst] ADD CONSTRAINT [PK__RJTEst__D86D1816BE9882BC] PRIMARY KEY CLUSTERED  ([a], [b]) ON [PRIMARY]
-GO
